enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+through the consoleclidownloads mechansism).
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs available for OpenShift and related services so that I can discover, install and list them
+The user will invoke the following commands:
+
+* `oc tools list` will access the connected OpenShift cluster, if connected,  and will retrieve information about available CLIs (ClusterCLI CRs)
+* `oc tools install odo` will install `odo` to a user's home directory, as a standalone binary.
+
+##### Example
+```text
+$ oc tools list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
+odo                  OpenShift Developer CLI     1.0          Not Installed
+kn                   Knative CLI                 1.0          Not Installed
+tkn                  Tekton CLI                  1.0          Not Installed
+
+```
+
+#### Story 2
+
+As a user, I want to access various CLIs from the OpenShift Console Web Terminal so that I can use the
+services available in the cluster. The CLIs versions should match the those of the operators deployed
+on the cluster.
+
+Currently, users can access `consoleclidownloads` custom resources for helm, oc, and odo either from oc with `oc get consoleclidownloads` or through
+the `downloads -n openshift-console` route. This access will still be available, through the updated ClusterCLI custom resources. The ClusterCLI custom
+resources can also be accessed through the console by searching for `ClusterCLI` as you would any other resource. The CLI versions that are compatible
+with a given release payload will in this way be available to the Console Web Terminal. CLI artifacts will continue to be served via a webserver for users to download
+from a route as they are today with the console downloads route. Today, only oc binaries are available with the console downloads route. The other consolclidownloads
+custom resources offer the external links to the user. Other CLIs can provide an artifact image to be shipped with the release payload if they wish to offer binaries
+served in this way from an imagestream located within the cluster registry. The CLIS currently available as `consoleclidownloads`, `oc, helm, and odo`, will be
+included in the reference implementation of this enhancement.
+
+The deployment of the CRD controller will have an API accessible to the console for future CLI list and download integration in the future.
+
+#### Story 3
+
+As owner of a CLI tool, I want to publish my CLI tool to users of the cluster. I need to create a CR for my tool, or provide the required information about my tool
+to a cluster-admin for the creation of a CR:
+* Name
+* Short description
+* Long description
+* Caveats
+* Homepage
+* Version
+* Platform/architecture
```

Comment:
This field implies that perhaps I could have 2 CRs for the same command for different architectures? Is the name of the CR significant, or does `tool install` look only at the path field?

Reply:
I was going to put a note there for clarification, but thought it better to be in a design details section: A single CR supports multiple versions, and each version supports multiple platforms (os/arch). The name of the CR is used for `tool install`.